### ISSUE TITLE
'url' should be defaulted in Response object according to fetch spec

### ISSUE THREAD

**Issue by Retro64** [bitinn/node-fetch#635]
Based on: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/35303#issuecomment-490975980

According to the fetch specification (https://fetch.spec.whatwg.org/#response-class), the url property should be defaulted in the Response object:

> The url attribute’s getter, when invoked, must return the empty string if the context object’s response’s URL is null and the context object’s response’s URL, serialized with the exclude-fragment flag set, otherwise. [URL]

Fix should be quite easy in https://github.com/bitinn/node-fetch/blob/master/src/response.js#L40

```
url: opts.url ? opts.url : '',
```

or

```
url: typeof opts.url === 'string' ? opts.url : '',
```
or even a REGEX checking opts.url for a url pattern.

I might provide a PR

**Comment by jimmywarting** [comment]
We are planning on using the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL)

```js
url: new URL(opts.url, baseUrl)
```

**Comment by Retro64** [comment]
@jimmywarting  Ok - if there is already a plan, I will keep my fingers away from creating a PR. Thanks for the fast reaction

**Comment by bitinn** [comment]
This is fixed and release as v2.6.0, we make sure default `Response.url` will not return `null` but return empty string instead, none of our tests rely on this and hopefully no one rely on this behavior neither.
